internal/http/services/sciencemesh/wayf.go

```diff
@@ -0,0 +1,256 @@
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+package sciencemesh
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/opencloud-eu/reva/v2/internal/http/services/ocmd"
+	"github.com/opencloud-eu/reva/v2/internal/http/services/reqres"
+	"github.com/opencloud-eu/reva/v2/pkg/appctx"
+)
+
+type wayfHandler struct {
+	federations []Federation
+}
+
+type Federation struct {
+	Federation string             `json:"federation"`
+	Servers    []FederationServer `json:"servers"`
+}
+
+// FederationServer represents a single provider with discovery info
+type FederationServer struct {
+	DisplayName        string `json:"displayName"`
+	URL                string `json:"url"`
+	InviteAcceptDialog string `json:"inviteAcceptDialog,omitempty"`
+}
+
+// federationFile is the on-disk structure without inviteAcceptDialog
+type federationFile struct {
+	Federation string                 `json:"federation"`
+	Servers    []federationServerFile `json:"servers"`
+}
+
+type federationServerFile struct {
+	DisplayName string `json:"displayName"`
+	URL         string `json:"url"`
+}
+
+type DiscoverRequest struct {
+	Domain string `json:"domain"`
+}
+
+type DiscoverResponse struct {
+	InviteAcceptDialog string `json:"inviteAcceptDialog"`
+}
+
+func (h *wayfHandler) init(c *config) error {
+	log := appctx.GetLogger(context.Background())
+
+	log.Debug().Str("file", c.FederationsFile).Msg("Initializing WAYF handler with federations file")
+
+	data, err := os.ReadFile(c.FederationsFile)
+	if err != nil {
+		if os.IsNotExist(err) {
+			log.Warn().Str("file", c.FederationsFile).Msg("Federations file not found, starting with empty list")
+			h.federations = []Federation{}
+			return nil
+		}
+		log.Error().Err(err).Str("file", c.FederationsFile).Msg("Failed to read federations file")
+		return err
+	}
+
+	var fileData []federationFile
+	if err := json.Unmarshal(data, &fileData); err != nil {
+		log.Error().Err(err).Str("file", c.FederationsFile).Msg("Failed to parse federations file")
+		return err
+	}
+
+	log.Debug().Int("federations_count", len(fileData)).Msg("Loaded federations from file")
+
+	// Create OCM client for discovery
+	ocmClient := ocmd.NewClient(10*time.Second, false)
+	log.Debug().Msg("Created OCM client for discovery")
+
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+	defer cancel()
+
+	// Discover each server and populate inviteAcceptDialog
+	h.federations = []Federation{}
+	discoveryErrors := 0
+	validServersCount := 0
+
+	for _, fed := range fileData {
+		log.Debug().Str("federation", fed.Federation).Int("servers_count", len(fed.Servers)).Msg("Processing federation")
+		var validServers []FederationServer
+
+		for _, srv := range fed.Servers {
+			if srv.DisplayName == "" || srv.URL == "" {
+				log.Warn().Str("federation", fed.Federation).
```

Comment:
Since this is only logged once on startup consider making this just info level, or maybe even debug. Though in my opinion it is useful to have this written out on startup. Admin can silence it by using log lever warn or error.

```suggestion
				log.Info().Str("federation", fed.Federation).
```